Assets/BossRoom/Scripts/Client/UI/PopupManager.cs

```diff
@@ -0,0 +1,90 @@
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+namespace Unity.Multiplayer.Samples.BossRoom.Visual
+{
+    /// <summary>
+    /// Handles the display of Popup messages. Instantiates and reuses popup panel prefabs to allow displaying multiple
+    /// messages in succession.
+    /// </summary>
+    public class PopupManager : MonoBehaviour
+    {
+        [SerializeField]
+        GameObject m_PopupPanelPrefab;
+
+        List<PopupPanel> m_PopupPanels = new List<PopupPanel>();
+
+        static PopupManager s_Instance;
```

Comment:
This is a good candidate for being turned into singleton with our DI.

Reply:
Since this is defined in the Client assembly, I cannot bind it inside ApplicationController, however, so it should either be moved to the Shared assembly or it should be bound in all the ClientGameStates. If the latter makes more sense, it would need to wait until PR #579 is merged to avoid conflicts